### PR TITLE
Upcoming appointments list

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
Upcoming appointments list page for homepage refresh.

I also moving some of the setup code around so that further appointment list page routes will live in the appointment-list folder, like for our other application sections.

The bulk of the code is in unit tests and copied and pasted from the current list component.

## Testing done
Local and unit testing

## Screenshots
Screen Shot 2021-01-15 at 11 34 23 AM
Screen Shot 2021-01-15 at 11 34 06 AM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/AppointmentsPage/AppointmentListItem.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+function VAFacilityName({ facility }) {
+  if (facility) {
+    return <>{facility.name}</>;
+  }
+
+  return 'VA appointment';
+}
+
+export default function AppointmentListItem({ appointment, facility }) {
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/AppointmentsPage/AppointmentListItem.jsx:73]: Most important/new stuff in the ticket is in here.

File: src/applications/vaos/appointment-list/components/AppointmentsPage/AppointmentsPageV2.jsx [hunk 2]
```diff
@@ -129,74 +116,42 @@ function AppointmentsPageV2({
...
-              options={options}
-              onChange={onChange}
-              aria-labelledby="options"
-            />
-            {routes}
-          </>
-        </>
-      )}
-    </PageLayout>
+      {expressCare.useNewFlow &&
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/AppointmentsPage/AppointmentsPageV2.jsx:119]: Further cleaned up some EC logic in here that I didn't notice before. We always have to show EC requests somehow, so the base feature flag doesn't make sense in the context of the new homepage.

File: src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx [hunk 3]
```diff
@@ -110,7 +109,7 @@ function AppointmentsPage({
   );
 
   return (
-    <PageLayout>
+    <>
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/AppointmentsPage/index.jsx:112]: Moved PageLayout to the appointment list section component

File: src/applications/vaos/appointment-list/index.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Switch, Route } from 'react-router-dom';
+import { selectFeatureHomepageRefresh } from '../redux/selectors';
+import PageLayout from './components/AppointmentsPage/PageLayout';
+import AppointmentsPageV2 from './components/AppointmentsPage/AppointmentsPageV2';
+import AppointmentsPage from './components/AppointmentsPage/index';
+
+function AppointmentListSection({ featureHomepageRefresh }) {
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/index.jsx:9]: Following the pattern of the other sections, other detail page routes will go in here

File: src/applications/vaos/services/appointment/index.js [hunk 5]
```diff
@@ -337,6 +337,39 @@ export function isUpcomingAppointmentOrRequest(appt) {
...
+    return (
+      !appt.vaos.isPastAppointment &&
+      !FUTURE_APPOINTMENTS_HIDDEN_SET.has(appt.description) &&
+      apptDateTime.isValid() &&
+      apptDateTime.isBefore(moment().add(13, 'months'))
+    );
+  }
+
+  return (
+    appt.vaos.isExpressCare &&
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/index.js:362]: EC logic is kind of guessing, I think we need to sit down with Peter and get a better understanding of how EC is handled in the refresh.

**Review by narin — APPROVED** [review]
LGTM! Nice work.

File: src/applications/vaos/appointment-list/redux/selectors.js [hunk 6]
```diff
@@ -115,6 +118,23 @@ export const selectFutureAppointments = createSelector(
   },
 );
 
+export const selectUpcomingAppointments = createSelector(
+  state => state.appointments.pending,
```
Comment (narin) [inline src/applications/vaos/appointment-list/redux/selectors.js:122]: May want to add a comment here making note that we are grabbing pending only for Express Care appts.

File: src/platform/testing/local-dev-mock-api/common.js [hunk 7]
```diff
@@ -77,6 +77,7 @@ const responses = {
         { name: 'vaOnlineSchedulingFlatFacilityPage', value: true },
         { name: 'vaOnlineSchedulingProviderSelection', value: true },
         { name: 'vaOnlineSchedulingCheetah', value: true },
+        { name: 'vaOnlineSchedulingHomepageRefresh', value: true },
```
Comment (jbalboni) [inline src/platform/testing/local-dev-mock-api/common.js:80]: @department-of-veterans-affairs/frontend-review-group This is the platform change, just to add a new feature flag to the default mock data.

**Review by U-DON — APPROVED** [review]

**Pull request merged**